test/integration/singlecluster/scheduler/scheduler_test.go

```diff
@@ -1251,9 +1251,9 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			util.WaitForNextSecondAfterCreation(pWl2)
 			util.ExpectWorkloadsToBePending(ctx, k8sClient, dWl1, pWl2)
 
-			ginkgo.By("Finishing one workload for prod ClusterQueue")
-			util.FinishWorkloads(ctx, k8sClient, pWl1)
+			ginkgo.By("Finishing first workload for prod ClusterQueue")
 			util.ExpectWorkloadsToBePending(ctx, k8sClient, dWl1, pWl2)
```

Comment:
We should first check if it’s pending, and then finish the first workload.

Reply:
I believe this makes sense, but could you describe in more detail why the previous order is problematic?

Reply:
After finishing the first workload, it should admit the second prod workload. There is enough time to admit it before we check for pending.

Reply:
Actually, we can remove this line at all. Because we are testing it above.

Reply:
I see, but on line 1252 we have `util.ExpectWorkloadsToBePending(ctx, k8sClient, dWl1, pWl2)` , so now line 1255 looks redundant. Something is off.

Reply:
Removed this line.